asyncapi/asb/caller.bal

```diff
@@ -51,152 +51,43 @@ public class Caller {
-    #                    integer assigned to a message as it is accepted and stored by the broker and functions as
-    #                    its true identifier.
-    # + return - An `asb:Error` if failed to receive deferred message, a Message record if successful or else `()`
-    public isolated function receiveDeferred(@display {label: "Sequence Number of the deferred message"} 
-                                             int sequenceNumber) 
-                                             returns @display {label: "Deferred Message record"} Message|Error? {
-        Message? message = check receiveDeferredMessage(self, sequenceNumber);
-        if message is Message {
-            check self.mapContentType(message);
-            return message;
-        }
-        return;
-    }
-
-    # The operation renews lock on a message in a queue or subscription based on messageLockToken.
-    #
-    # + message - Message record
-    # + return - An `asb:Error` if failed to renew message or else `()`
-    public isolated function renewLock(@display {label: "Message record"} Message message) returns Error? {
-        if message?.lockToken.toString() != DEFAULT_MESSAGE_LOCK_TOKEN {
-            return renewLockMessage(self, message?.lockToken.toString());
-        }
-        return error Error("Failed to renew lock on message with ID " + message?.messageId.toString());
-    }
-
-    # Set the prefetch count of the receiver.
-    # Prefetch speeds up the message flow by aiming to have a message readily available for local retrieval when and
-    # before the application asks for one using Receive. Setting a non-zero value prefetches PrefetchCount
-    # number of messages. Setting the value to zero turns prefetch off. For both PEEKLOCK mode and
-    # RECEIVEANDDELETE mode, the default value is 0.
-    #
-    # + prefetchCount - The desired prefetch count.
-    # + return - An `asb:Error` if failed to renew message or else `()`
-    public isolated function setPrefetchCount(@display {label: "Prefetch count"} int prefetchCount) returns Error? {
-        return setPrefetchCountMessage(self, prefetchCount);
-    }
-
-    isolated function mapContentType(Message message) returns Error? {
-        match message?.contentType {
-            TEXT => {
-                message.body = check nativeGetTextContent(<byte[]>message.body);
-            }
-            JSON => {
-                message.body = check nativeGetJSONContent(<byte[]>message.body);
-            }
-            XML => {
-                message.body = check nativeGetXMLContent(<byte[]>message.body);
-            }
-            BYTE_ARRAY => {
-                message.body = <byte[]>message.body;
-            }
-            _ => {
-                message.body = message.body.toString();
-            }
-        }
+    public isolated function defer(@display {label: "Message record"} Message message)
```

Comment:
If user uses it as 
message.differ() -- > Name differ is fine 

If he uses it as

caller.differ() --> maybe better rename to differMessage() ?

Reply:
caller.defer(message) is the correct way user should use. 
Hw, we can change to caller.differMessage(message) , and this is what in SDK  - public void defer(DeferOptions options)

Reply:
This `caller ` only has methods operated on a message? If so, it would be fine as message goes as an input param.

Reply:
yes.

Reply:
Ack